File: web/Process.js
class Process {
    constructor(PID, arrivalTime, burstTime) {
        this.PID = PID;
        this.arrivalTime = arrivalTime;
        this.burstTime = burstTime;
        this.remainingTime = burstTime;
        this.waitingTime = 0;
        this.turnaroundTime = 0;
        this.completionTime = 0;
        this.responseTime = -1;
        this.completed = false;
        this.isStarted = false;
    }

    getPID() {
        return this.PID;
    }

    getArrivalTime() {
        return this.arrivalTime;
    }

    getBurstTime() {
        return this.burstTime;
    }

    getRemainingTime() {
        return this.remainingTime;
    }

    getWaitingTime() {
        return this.waitingTime;
    }

    getTurnaroundTime() {
        return this.turnaroundTime;
    }

    getCompletionTime() {
        return this.completionTime;
    }

    getResponseTime() {
        return this.responseTime;
    }

    isCompleted() {
        return this.completed;
    }

    isStarted() {
        return this.isStarted;
    }

    setRemainingTime(remainingTime) {
        this.remainingTime = remainingTime;
    }

    setWaitingTime(waitingTime) {
        this.waitingTime = waitingTime;
    }

    setTurnaroundTime(turnaroundTime) {
        this.turnaroundTime = turnaroundTime;
    }

    setCompletionTime(completionTime) {
        this.completionTime = completionTime;
    }

    setResponseTime(responseTime) {
        this.responseTime = responseTime;
    }

    setCompleted(completed) {
        this.completed = completed;
    }

    setStarted(started) {
        this.isStarted = started;
    }
}

File: src/Process.java
public class Process {
    private int PID;
    private int arrivalTime;
    private int burstTime;
    private int remainingTime;
    private int waitingTime;
    private int turnaroundTime;
    private int completionTime;
    private int responseTime;
    private boolean completed;
    private boolean isStarted;

    public Process(int PID, int arrivalTime, int burstTime) {
        this.PID = PID;
        this.arrivalTime = arrivalTime; 
        this.burstTime = burstTime;
        this.remainingTime = burstTime;
        this.responseTime = -1;
        this.isStarted = false;
        this.completed = false;
    }

    public int getPID() {
        return PID;
    }

    public int getArrivalTime() {
        return arrivalTime;
    }

    public int getBurstTime() {
        return burstTime;
    }

    public int getRemainingTime() {
        return remainingTime;
    }

    public int getWaitingTime() {
        return waitingTime;
    }

    public int getTurnaroundTime() {
        return turnaroundTime;
    }

    public int getCompletionTime() {
        return completionTime;
    }

    public int getResponseTime() {
        return responseTime;
    }

    public boolean isCompleted() {
        return completed;
    }

    public boolean isStarted() {
        return isStarted;
    }

    public void setRemainingTime(int remainingTime) {
        this.remainingTime = remainingTime;
    }

    public void setWaitingTime(int waitingTime) {
        this.waitingTime = waitingTime;
    }

    public void setTurnaroundTime(int turnaroundTime) {
        this.turnaroundTime = turnaroundTime;
    }

    public void setCompletionTime(int completionTime) {
        this.completionTime = completionTime;
    }

    public void setResponseTime(int responseTime) {
        this.responseTime = responseTime;
    }

    public void setCompleted(boolean completed) {
        this.completed = completed;
    }

    public void setStarted(boolean started) {
        this.isStarted = started;
    }
}